dart/lib/src/protocol/access_aware_map.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+
+@internal
+class AccessAwareMap<String, V> extends MapBase<String, V> {
+  AccessAwareMap(this._map);
+
+  final Map<String, V> _map;
+  final Set<String> _accessedKeys = {};
+
+  Set<String> get accessedKeys => _accessedKeys;
+
+  @override
+  V? operator [](Object? key) {
+    if (key is String) {
+      _accessedKeys.add(key);
+    }
+    return _map[key];
+  }
+
+  @override
+  void operator []=(String key, V value) {
+    _map[key] = value;
+  }
```

Comment:
Even though this code is trivial, it may be worth to add simple tests

Reply:
added tests for all the mapbase functionality